Sources/TuistCache/Mappers/CacheGraphMutator.swift

```diff
@@ -123,31 +125,41 @@ class CacheGraphMutator: CacheGraphMutating {
 
     // swiftlint:disable:next function_body_length
     fileprivate func mapDependencies(
-        _ dependencies: Set<GraphDependency>,
-        graph: Graph,
+        for dependency: GraphDependency,
+        graphTraverser: GraphTraverser,
         graphDependencies: inout [GraphDependency: Set<GraphDependency>],
         precompiledArtifacts: [GraphTarget: AbsolutePath],
         sources: Set<String>,
         sourceTargets: inout Set<GraphTarget>,
         visitedPrecompiledArtifactPaths: inout [GraphTarget: VisitedArtifact?],
         loadedPrecompiledArtifacts: inout [AbsolutePath: GraphDependency]
     ) throws -> Set<GraphDependency> {
+        // Dependency already visited
+        guard graphDependencies[dependency] == nil else { return graphDependencies[dependency]! }
```

Comment:
I know it's an early exit and a guard wouild be more appropriate, but in this case I find this much clearer
```suggestion
        if let cachedGraphDependencies = graphDependencies[dependency] {
          return cachedGraphDependencies
        }
```